CMakeLists.txt

```diff
@@ -205,6 +205,9 @@ if(BUILD_TESTING)
     target_sources(ninja_test PRIVATE src/includes_normalize_test.cc src/msvc_helper_test.cc)
   endif()
   target_link_libraries(ninja_test PRIVATE libninja libninja-re2c)
+  if (MSVC AND "$ENV{VSCMD_ARG_TGT_ARCH}" STREQUAL "x86")
+    target_compile_options(ninja_test PRIVATE "/fsanitize=address")
```

Comment:
Adding /fsanitize=address is not related to adding long path awareness to Ninja.

Don't get me wrong, it should be added to the build, but it should be done as a completely separate PR.